[Recognizers-Text][Python] Add TimeZoneExtractor property

Konigstiger opened the pull request:
### Changes made:

We added or updated, where applicable, the property _TimeZoneExtractor_ for the TimeZone implementation. 

#### The classes affected were:
- **DateTimePeriodExtractorConfiguration**
- **MergedExtractorConfiguration**
- **TimePeriodExtractorConfiguration**
- **EnglishMergedExtractorConfiguration**

- **FrenchDateTimePeriodParserConfiguration** 
- **FrenchTimeParserConfiguration** 
- **FrenchTimePeriodParserConfiguration**

- **SpanishDateTimePeriodParserConfiguration**
- **SpanishTimeParserConfiguration** 
- **SpanishTimePeriodParserConfiguration**

Commit:
Add TimeZoneExtractor property to DateTimePeriodExtractorConfiguration

Changes:
Add TimeZoneExtractor property to DateTimePeriodExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to MergedExtractorConfiguration

Changes:
Add TimeZoneExtractor property to MergedExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to TimePeriodExtractorConfiguration

Changes:
Add TimeZoneExtractor property to TimePeriodExtractorConfiguration class

Commit:
Add TimeZoneExtractor property to EnglishMergedExtractorConfiguration

Changes:
Add TimeZoneExtractor property to EnglishMergedExtractorConfiguration class

Commit:
Add time_zone_parser property to FrenchDateTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
Add time_zone_parser property to FrenchTimeParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
Add time_zone_parser property to FrenchTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
Add time_zone_parser property to Spanish classes

Classes affected:
- SpanishTimePeriodParserConfiguration
- SpanishTimeParserConfiguration
- SpanishDateTimePeriodParserConfiguration

Changes:
- Add import for BaseTimeZoneParser
- Add property time_zone_parser
- Add initialization

Commit:
PEP8

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/french/datetimeperiod_parser_config.py:
```diff
@@ -64,6 +64,7 @@ def __init__(self, config: BaseDateParserConfiguration):
             FrenchDateTime.RelativeTimeUnitRegex)
         self._rest_of_date_time_regex = RegExpUtility.get_safe_reg_exp(
             FrenchDateTime.RestOfDateTimeRegex)
+        self._time_zone_parser = BaseTimeZoneParser()
```
This is not correct. This is C# initialization of this prop

`TimeZoneParser = config.TimeZoneParser;`

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/french/time_parser_config.py:
```diff
@@ -42,6 +47,7 @@ def __init__(self, config: BaseDateParserConfiguration):
 
         self._utility_configuration = config.utility_configuration
         self._numbers: Dict[str, int] = config.numbers
+        self._time_zone_parser = BaseTimeZoneParser()
```
Same as above

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/french/timeperiod_parser_config.py:
```diff
@@ -63,6 +68,7 @@ def __init__(self, config: BaseDateParserConfiguration):
             FrenchDateTime.TimeOfDayRegex)
         self._till_regex = RegExpUtility.get_safe_reg_exp(
             FrenchDateTime.TillRegex)
+        self._time_zone_parser = BaseTimeZoneParser()
```
Same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/spanish/datetimeperiod_parser_config.py:
```diff
@@ -55,6 +56,7 @@ def __init__(self, config: BaseDateParserConfiguration):
             SpanishDateTime.RelativeTimeUnitRegex)
         self._rest_of_date_time_regex = RegExpUtility.get_safe_reg_exp(
             SpanishDateTime.RestOfDateTimeRegex)
+        self._time_zone_parser = BaseTimeZoneParser()
```
Same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/spanish/time_parser_config.py:
```diff
@@ -42,6 +47,7 @@ def __init__(self, config: BaseDateParserConfiguration):
 
         self._utility_configuration = config.utility_configuration
         self._numbers: Dict[str, int] = config.numbers
+        self._time_zone_parser = BaseTimeZoneParser()
```
Same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/spanish/timeperiod_parser_config.py:
```diff
@@ -62,6 +67,7 @@ def __init__(self, config: BaseDateParserConfiguration):
             SpanishDateTime.TimeOfDayRegex)
         self._till_regex = RegExpUtility.get_safe_reg_exp(
             SpanishDateTime.TillRegex)
+        self._time_zone_parser = BaseTimeZoneParser()
```
Same

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_timeperiod.py:
```diff
@@ -69,6 +69,11 @@ def token_before_date(self) -> str:
 
+    @property
+    @abstractmethod
+    def time_zone_extractor(self) -> DateTimeParser:
```
Here the return type is not DateTimeParser or the property has the be renamed as _time_zone_parser_. Please, check it.

Konigstiger replied:
True!, It was DateTimeExtractor, thanks.

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_merged.py:
```diff
@@ -183,6 +183,11 @@ def ambiguity_filters_dict(self) -> {}:
 
+    @property
+    @abstractmethod
+    def time_zone_extractor(self) -> DateTimeParser:
```
Check the return type

Maihav commented on Python/libraries/recognizers-date-time/recognizers_date_time/date_time/base_datetimeperiod.py:
```diff
@@ -205,6 +205,11 @@ def suffix_regex(self) -> Pattern:
 
+    @property
+    @abstractmethod
+    def time_zone_extractor(self) -> DateTimeParser:
```
Check the return type

Commit:
Applied Feedback

Changes:
- Adjusted imports as necessary.
- Changed return type and initialization of time_zone_extractor.
